### PR TITLE
Dedupe some box anchoring code between legend.py and offsetbox.py.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by timhoffm — APPROVED** [review]
+/- optional changes.

On a more general note, it feels wrong that legend needs to explicitly handle locating. Shouldn't a legend be some subclass of an anchored box (I'm not very familiar with that part of the library, so I don't know which class exactly)?

File: lib/matplotlib/offsetbox.py [hunk 1]
```diff
@@ -1146,31 +1143,17 @@ def draw(self, renderer):
...
-                        C: "C"}
-
-        c = anchor_coefs[loc]
-
-        container = parentbbox.padded(-borderpad)
-        anchored_box = bbox.anchored(c, container=container)
-        return anchored_box.x0, anchored_box.y0
+def _get_anchored_bbox(loc, bbox, parentbbox, borderpad):
+    """
+    Return the position of the *bbox* anchored at the *parentbbox* with the
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/offsetbox.py:1149]: sure

File: lib/matplotlib/offsetbox.py [hunk 2]
```diff
@@ -1146,31 +1143,17 @@ def draw(self, renderer):
...
+    *loc* code with the *borderpad*.
+    """
+    c = _api.check_getitem({1: "NE", 2: "NW", 3: "SW", 4: "SE", 5: "E",
+                            6: "W", 7: "E", 8: "S", 9: "N", 10: "C"}, loc=loc)
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/offsetbox.py:1153]: I went for another way to fit everything in a single line (see comment).  This is private API and `loc` should normally have already been validated, so there should not be an observable API break.

**Comment by anntzer** [comment]
It's probably possible to make legend reuse more of the offsetbox code, but there'd need to be quite a bit of backcompat/surgery due to different attribute names, etc. so that'd be a separate project.

**Comment by QuLogic** [comment]
Did you not push the optional changes?

**Comment by anntzer** [comment]
oopsie.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**